project/UI/Carbon/Carbon.swift

```diff
@@ -2,71 +2,56 @@
-                                     .init(name: CarbonCommand.size.rawValue, placeholder: "", description: "export file size [1-5].", default: "2"),
-                                     .init(name: CarbonCommand.font.rawValue, placeholder: "", description: "carbon's font type.", default: "fira-code"),
-                                     .init(name: CarbonCommand.lines.rawValue, placeholder: "", description: "shows/hides lines of code [true | false].", default: "true"),
-                                     .init(name: CarbonCommand.watermark.rawValue, placeholder: "", description: "shows/hides the watermark [true | false].", default: "true"))
+    @ArgumentParser.Option(default: .dracula, help: "Carbon theme")
+    private var theme: CarbonStyle.Theme
     
-    return arguments(console: console)
-        .flatMap { (input, output, style) in
-            nef.Carbon.render(playgroundsAt: input, style: style, into: output)
-                      .provide(console)^
-                      .mapError { _ in .render() }
-                      .foldM({ _ in console.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
-                             { _ in console.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   }) }^
-        .reportStatus(in: console)
-        .foldM({ e in console.exit(failure: "\(e)")        },
-               { success in console.exit(success: success) })
-        .unsafeRunSyncEither()
+    @ArgumentParser.Option(default: .x2, help: "export file size [1-5]")
+    private var size: CarbonStyle.Size
+    
+    @ArgumentParser.Option(default: .firaCode, help: "Carbon font type")
+    private var font: CarbonStyle.Font
+    
+    @ArgumentParser.Option(name: .customLong("show-lines"), default: true, help: "Shows/hides lines of code [true | false]")
+    private var lines: Bool
+    
+    @ArgumentParser.Option(name: .customLong("show-watermark"), default: true, help: "Shows/hides the watermark [true | false]")
+    private var watermark: Bool
+    
+    
+    public func run() throws {
+        let style = CarbonStyle(background: CarbonStyle.Color(hex: background) ?? CarbonStyle.Color(default: background) ?? CarbonStyle.Color.nef,
+                                theme: theme,
+                                size: size,
+                                fontType: font,
+                                lineNumbers: lines,
+                                watermark: watermark)
+        
+        try nef.Carbon.render(playgroundsAt: project.url, style: style, into: output.url)
+                .provide(Console.default)^
+                .foldM({ _ in Console.default.exit(failure: "rendering Xcode Playgrounds from '\(self.project.path)'") },
+                       { _ in Console.default.exit(success: "rendered Xcode Playgrounds in '\(self.output.path)'")   })^
+                .unsafeRunSync()
```

Comment:
I'd try to make this as succinct as possible, something along the line of:

```swift
try nef.Carbon.render(...).provide(Console.default)^.unsafeRunSync()
```

Specially, the last `foldM` happening after providing the dependency (which it uses undercover) doesn't look good to me. My suggested option lets us write tests end-to-end, whereas this option will not let us test the last `foldM` instructions.

Reply:
done